apps/server/src/modules/tldraw/config.ts

```diff
@@ -24,6 +25,7 @@ const tldrawConfig = {
 	CONNECTION_STRING: tldrawConnectionString,
 	TLDRAW_PING_TIMEOUT: Configuration.get('TLDRAW__PING_TIMEOUT') as number,
 	TLDRAW_GC_ENABLED: Configuration.get('TLDRAW__GC_ENABLED') as boolean,
+	REDIS_URI: Configuration.has('REDIS_URI') ? (Configuration.get('REDIS_URI') as string) : 'redis://127.0.0.1:6379',
```

Comment:
I actually only need to set this for tests but I am not sure if there is a better way.
If I set this up in /config/test.json the feathers test will fail because it will try to connect to Redis and crash.
Cannot set it up in default.schema.json as there is some logic that works specifically if this env var is not defined.
Could not set this inside tests either as suite will not even start when env var is used but not defined.

Reply:
Either we make the redis required for the server (including local), or we have to make sure the server can start without it.
Like this, you will have undefined behavior. As you mentioned, there is logic that is accessing this value to decide stuff (to activate the autologout), and im not sure what will happen there if you have a value for the redis url without there actually being a redis behind it. Making the Redis required is a bigger decision, that should be taken into the arc chapter if its necessary.

Why does your suite not start without redis? it would also make sense to confirm the behavior of your code (that its disabled) when there is no redis, as that is a valid case at the moment. So I think we need to find a solution to this anyway.

Reply:
If redis is required for tldraw, I think its fine to fail (crash) during startup if tldraw is enabled and there is no redis URI.
You can make this clearer by adding a restriction to the schema, but not strictly necessary.

But that doesnt help for your test case. I am however not aware of any checks jest makes before running a testsuite, at least the beforeAll should run, and you should be able to mock the configuration before creating the test module... But tbh I havent tried that myself. If you are having trouble, please point me to a specific case then maybe I can investigate it a bit.

Reply:
wrote you a longer message in rocketchat with suggestions